crates/cext/src/circuit.rs

```diff
@@ -378,6 +381,59 @@ pub struct OpCounts {
     len: usize,
 }
 
+/// @ingroup QkCircuit
+/// Append an arbitrary unitary matrix to the circuit.
+/// The user passes a row-major array of interleaved real-imaginary pairs #[
+/// representing the unitary matrix.
+///
+/// @param curcuit      A pointer to the circuit to add the unitary to.
+/// @param num_qubits   The width of unitary gate.
+/// @param qubits       The pointer to array of quibit indices.
+/// The array length must be equal to num_qubits.
+/// @param matrix       The pointer to the row-major `(re, im)` doucle array///
```

Comment:
```suggestion
/// @param matrix       The pointer to the row-major `(re, im)` double array
```

Besides the typo fix suggestion above did you want to make some comments about striding here? I think just defining unit striding is fine. But we should be pretty explicit about the expected memory layout to not lead to surprises. An example of how the array is constructed in C would be really good so you can say "this functions expects if you create an array like this or in some other way with the same memory layout".